lamindb/curators/__init__.py

```diff
@@ -3336,22 +3262,27 @@ def save_artifact(
     from ..models.artifact import add_labels, data_is_anndata, data_is_mudata
 
     if artifact is None:
-        if data_is_anndata(data):
-            artifact = Artifact.from_anndata(
+        if isinstance(data, pd.DataFrame):
+            artifact = Artifact.from_df(
                 data, description=description, key=key, revises=revises, run=run
             )
-        elif isinstance(data, pd.DataFrame):
-            artifact = Artifact.from_df(
+        elif data_is_anndata(data):
+            artifact = Artifact.from_anndata(
                 data, description=description, key=key, revises=revises, run=run
             )
         elif data_is_mudata(data):
             artifact = Artifact.from_mudata(
-                data,
-                description=description,
-                key=key,
-                revises=revises,
-                run=run,
+                data, description=description, key=key, revises=revises, run=run
+            )
+        elif data_is_spatialdata(data):
+            artifact = Artifact.from_spatialdata(
+                data, description=description, key=key, revises=revises, run=run
             )
+        else:
+            raise InvalidArgument(  # pragma: no cover
+                "data must be one of pd.Dataframe, AnnData, MuData, SpatialData."
+            )
+
     artifact.schema = schema
```

Comment:
This was my fault, could you move the schema FK assignment to the end?

Keep the `artifact.save()` here.

Add `artifact.schema = schema; artifact.save()` at the end of `save_artifact`.